### PR TITLE
 Reissuance: Accompanying certificates screen (EXPOSUREAPP-13068)

### PR DESCRIPTION
**Pull request by chiljamgossow**
In case the certificates for reissuance have accompanying certificates, there is a new tile that leads to a new screen.
Certificates are sorted and duplicates removed.

https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13068

https://www.figma.com/file/KuDEcMSREg3VhZRhC3FXTY/COVID19_Master_2.20?node-id=49861%3A66652

screenshots will come with a new PR

How to test: scan all of them
cwa dcc series -e cf --preset 18 -o

### CODE REVIEW

**Review by KathrinLuecke — DISMISSED** [review]
UA approved

**Review by SamuraiKek — DISMISSED** [review]
LGTM. Also tested on device and looks nice.

**Review by schauersbergern — DISMISSED** [review]
Works as expected

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/acccerts/DccReissuanceAccCertsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as DccReissuanceAccCertsViewModel.Factory
+            factory.create(
+                personIdentifierCode = args.personIdentifierCode,
+            )
+        }
+    )
+
+    private val dccReissuanceAdapter = DccReissuanceAdapter()
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/acccerts/DccReissuanceAccCertsFragment.kt:35]: Memory Leak!

The adapter will outlive the view, but holds a reference to the list internally, so the view cannot be destroyed. This even gets worse, when the fragment is kept on the backstack..

**Review by BMItr — CHANGES_REQUESTED** [review]
Please check the comment. (memory leak)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/acccerts/DccReissuanceAccCertsFragment.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/acccerts/DccReissuanceAccCertsFragment.kt:35]: This is copy paste from the consent view. Can you be more precise? How would the adapter outlive the view? How do you think should it be done?

Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/acccerts/DccReissuanceAccCertsFragment.kt]: you last change is sufficent.

**Review by BMItr — DISMISSED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/acccerts/DccReissuanceAccCertsFragment.kt (hunk 1 above)
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/acccerts/DccReissuanceAccCertsFragment.kt]: maybe you can change it also in the fragment where you copy pasted it?

File: Corona-Warn-App/src/main/res/values-de/dcc_reissuance_strings.xml [hunk 2]
```diff
@@ -27,9 +27,19 @@
...
     <!-- XTXT: DCC Reissuance deletion hint-->
     <string name="dcc_reissuance_deletion_hint">Das alte Zertifikat wird automatisch in den Papierkorb verschoben und nach 30 Tagen gelöscht. Das Zertifikat kann innerhalb der 30 Tage jederzeit wiederhergestellt werden.</string>
     <!-- XTXT: DCC Reissuance update hint-->
     <string name="dcc_reissuance_update_hint">Falls Sie hier abbrechen, können Sie die Aktualisierung jederzeit in Ihrer Zertifikatsübersicht anfordern.</string>
+    <!-- XTXT: DCC Reissuance accompanying certificates tile -->
+    <string name="dcc_reissuance_accompanying_certificates">Zusätzlich übermittelte Zertifikate</string>
+    <!-- XTXT: DCC Reissuance accompanying certificates content -->
+    <string name="dcc_reissuance_acc_certs_content">Folgende Zertifikate werden bei der Erneuerung als Zusatzinformation übermittelt, damit die bisherige Abfolge der Impfungen und Genesungen im neuen Zertifikat aufgenommen werden kann. Diese Zertifikate verbleiben unverändert auf Ihrem Gerät.</string>
```
Comment (KathrinLuecke) [inline Corona-Warn-App/src/main/res/values-de/dcc_reissuance_strings.xml:38]:
```suggestion
    <string name="dcc_reissuance_acc_certs_content">Folgende Zertifikate werden bei der Erneuerung als Zusatzinformation übermittelt, damit die bisherige Abfolge der Impfungen und Genesungen im neuen Zertifikat aufgenommen werden kann. Diese Zertifikate werden nicht erneuert.</string>
```

**Review by KathrinLuecke — CHANGES_REQUESTED** [review]

**Comment by mtwalli** [comment]
@SamuraiKek  @schauersbergern  @BMItr  Please recheck ?

**Review by BMItr — APPROVED** [review]
reapproved

**Review by KathrinLuecke — APPROVED** [review]
UA approved

**Review by schauersbergern — APPROVED** [review]

**Pull request merged**